bundles/SimpleBackendSearchBundle/src/Command/SearchBackendReindexCommand.php

```diff
@@ -46,62 +45,51 @@ protected function execute(InputInterface $input, OutputInterface $output): int
     {
         // clear all data
         $db = \Pimcore\Db::get();
-        $db->executeQuery('TRUNCATE `search_backend_data`;');
 
-        $elementsPerLoop = 100;
         $types = ['asset', 'document', 'object'];
 
         foreach ($types as $type) {
-            $baseClass = Service::getBaseClassNameForElement($type);
-            $listClassName = '\\Pimcore\\Model\\' . $baseClass . '\\Listing';
-            $list = new $listClassName();
-            if (method_exists($list, 'setUnpublished')) {
-                $list->setUnpublished(true);
-            }
-
-            if (method_exists($list, 'setObjectTypes')) {
-                $list->setObjectTypes([
-                    DataObject\AbstractObject::OBJECT_TYPE_OBJECT,
-                    DataObject\AbstractObject::OBJECT_TYPE_FOLDER,
-                    DataObject\AbstractObject::OBJECT_TYPE_VARIANT,
-                ]);
-            }
-
-            $elementsTotal = $list->getTotalCount();
-
-            for ($i = 0; $i < (ceil($elementsTotal / $elementsPerLoop)); $i++) {
-                $list->setLimit($elementsPerLoop);
-                $list->setOffset($i * $elementsPerLoop);
-
-                $this->output->writeln(
-                    'Processing ' .$type . ': ' . ($list->getOffset() + $elementsPerLoop) . '/' . $elementsTotal
-                );
-
-                $elements = $list->load();
-                foreach ($elements as $element) {
-                    try {
-                        //process page count, if not exists
-                        if (
-                            $element instanceof Asset\Document &&
-                            !$element->getCustomSetting('document_page_count') &&
-                            $element->processPageCount()
-                        ) {
-                            $this->saveAsset($element);
-                        }
+            $elementIds = array_column(
+                $db->fetchAllAssociative(
+                    'SELECT elements.id
+                    FROM `'.$type.'s` elements
+                    LEFT JOIN search_backend_data ON elements.id=search_backend_data.id
+                        AND search_backend_data.mainType=?
+                    WHERE search_backend_data.id IS NULL
+                        OR search_backend_data.modificationDate < elements.modificationDate',
+                    [$type]
+                ),
+                'id'
+            );
```

Comment:
This would be the same, no?

```suggestion
            $elementIds = $db->fetchFirstColumn(
                'SELECT elements.id
                FROM `'.$type.'s` elements
                LEFT JOIN search_backend_data ON elements.id=search_backend_data.id
                    AND search_backend_data.mainType=?
                WHERE search_backend_data.id IS NULL
                    OR search_backend_data.modificationDate < elements.modificationDate',
                [$type]
            );
```